### ISSUE TITLE
Stats: Refactor notices component for increased clarity

### ISSUE THREAD

**Issue by jsnmoon** [Automattic/wp-calypso#80472]
Capturing this follow-up task from https://github.com/Automattic/wp-calypso/pull/80462#issue-1845882231:
> We'll refactor the notices in follow up PRs to make the logic clearer.

cc @kangzj

```[tasklist]
### Tasks
- [x] Remove Opt-out & feedback notice & legacy notices ( https://github.com/Automattic/wp-calypso/pull/80480 )
- [ ] https://github.com/Automattic/wp-calypso/issues/80951
```

**Comment by jsnmoon** [comment]
@kangzj, what are your thoughts on deprecating both the opt-out and feedback notices?

Here’s my reasoning:

- The opt-out notice is being shown unnecessarily to new Jetpack customers visiting the Odyssey Stats dashboard for the first time.
- The opt-out notice has served its purpose of informing our customers that the older stats dashboard still exists.
- We don’t have anyone monitoring the survey results as closely as we used to, so it’s no longer a viable place for customers to send us feedback.

**Comment by kangzj** [comment]
Thanks for bring this to attention @jsnmoon 

> The opt-out notice is being shown unnecessarily to new Jetpack customers visiting the Odyssey Stats dashboard for the first time.
The opt-out notice has served its purpose of informing our customers that the older stats dashboard still exists.

Makes sense. The opt-out notice has served its mission and is time to retire.

> We don’t have anyone monitoring the survey results as closely as we used to, so it’s no longer a viable place for customers to send us feedback.

I would lean to remove it too, becuase it was focused on the feedback from the old Stats to new Stats, unless @ederrengifo or @madebynoam thinks otherwise.

**Comment by kangzj** [comment]
closed with #80949
